src/gax/src/client_builder.rs

```diff
@@ -386,6 +386,20 @@ impl<F, Cr> ClientBuilder<F, Cr> {
 pub mod internal {
     use super::*;
 
+    /// Options for instrumenting client behavior, primarily for tracing and metrics.
+    #[derive(Clone, Debug, PartialEq, Eq)]
+    #[non_exhaustive]
+    pub struct InstrumentationOptions {
```

Comment:
naming nit: s/Options/Config/ ?

unless this will grow to include option-like things.